src/analysis_context.hpp

```
@@ -2,25 +2,43 @@
+/// instance. Domain code reaches for the global `variable_registry` directly.
 struct AnalysisContext {
-    const ProgramInfo& program_info;
-    const ebpf_verifier_options_t& options;
-    const ebpf_platform_t& platform;
+    Program program;
+    ebpf_verifier_options_t options;
+
+    AnalysisContext(Program p, ebpf_verifier_options_t o) : program(std::move(p)), options(std::move(o)) {}
+
+    const ProgramInfo& program_info() const { return program.info(); }
+    const ebpf_platform_t& platform() const { return *program.info().platform; }
```

Clang (14.0.6)

[warning] function 'platform' should be marked [[nodiscard]]

(modernize-use-nodiscard)